<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <property name="spacing">
       <number>5</number>
      </property>
      <property name="leftMargin">
       <number>10</number>
      </property>
      <property name="topMargin">
       <number>10</number>
      </property>
      <property name="rightMargin">
       <number>10</number>
      </property>
      <property name="bottomMargin">
       <number>10</number>
      </property>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string/>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="1" column="0">
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>Show:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="lbl_coord">
           <property name="text">
            <string>Coordinate system:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QComboBox" name="cb_coord">
           <item>
            <property name="text">
             <string>Cartesian</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Spherical</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="1" column="2">
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QCheckBox" name="cb_axes">
             <property name="text">
              <string>Axes</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="cb_sphere">
             <property name="text">
              <string>Sphere</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_6">
        <property name="title">
         <string>Field sequence</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_19">
         <item row="3" column="2">
          <widget class="QDoubleSpinBox" name="coord3_e">
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="prefix">
            <string/>
           </property>
           <property name="decimals">
            <number>4</number>
           </property>
           <property name="minimum">
            <double>-1000.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1000.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>0.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="lbl_coord2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>y</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="coord1">
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="prefix">
            <string/>
           </property>
           <property name="decimals">
            <number>4</number>
           </property>
           <property name="minimum">
            <double>-1000.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1000.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="lbl_coord1">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>x</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="coord2">
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="prefix">
            <string/>
           </property>
           <property name="decimals">
            <number>4</number>
           </property>
           <property name="minimum">
            <double>-1000.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1000.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="label_19">
           <property name="text">
            <string>Last field</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="label_24">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>First field</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QDoubleSpinBox" name="coord1_e">
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="prefix">
            <string/>
           </property>
           <property name="decimals">
            <number>4</number>
           </property>
           <property name="minimum">
            <double>-1000.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1000.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>0.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="lbl_coord3">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>z</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QDoubleSpinBox" name="coord2_e">
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="prefix">
            <string/>
           </property>
           <property name="decimals">
            <number>4</number>
           </property>
           <property name="minimum">
            <double>-1000.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1000.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="coord3">
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="prefix">
            <string/>
           </property>
           <property name="decimals">
            <number>4</number>
           </property>
           <property name="minimum">
            <double>-1000.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1000.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>0.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_7">
        <property name="title">
         <string>Sweep</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_20">
         <item row="1" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Wait after</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignJustify|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QCheckBox" name="cb_shortest">
           <property name="text">
            <string>Shortest</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="PyScientificSpinBox" name="waitAfter">
           <property name="suffix">
            <string>s</string>
           </property>
           <property name="value">
            <double>0.020000000000000</double>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_18">
           <property name="text">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Nb pts&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="nbpts">
           <property name="maximum">
            <number>500000</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="PyScientificSpinBox" name="rampRate"/>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Ramp rate</string>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QPushButton" name="rampRateSave">
           <property name="toolTipDuration">
            <number>10000</number>
           </property>
           <property name="text">
            <string>Set ramp rate</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>Current field</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="1" column="0">
          <widget class="QLabel" name="label_3">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>y</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>x</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_4">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>z</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLCDNumber" name="lcdx">
           <property name="font">
            <font>
             <family>MS Shell Dlg 2</family>
            </font>
           </property>
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="digitCount">
            <number>10</number>
           </property>
           <property name="segmentStyle">
            <enum>QLCDNumber::Flat</enum>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLCDNumber" name="lcdy">
           <property name="digitCount">
            <number>10</number>
           </property>
           <property name="segmentStyle">
            <enum>QLCDNumber::Flat</enum>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLCDNumber" name="lcdz">
           <property name="digitCount">
            <number>10</number>
           </property>
           <property name="segmentStyle">
            <enum>QLCDNumber::Flat</enum>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QPushButton" name="btn_read_field">
           <property name="text">
            <string>Read</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="PyScientificSpinBox" name="sp_targetx"/>
         </item>
         <item row="1" column="2">
          <widget class="PyScientificSpinBox" name="sp_targety"/>
         </item>
         <item row="2" column="2">
          <widget class="PyScientificSpinBox" name="sp_targetz"/>
         </item>
         <item row="3" column="2">
          <widget class="QPushButton" name="btn_goto">
           <property name="text">
            <string>Go to</string>
           </property>
          </widget>
         </item>
         <item row="4" column="2">
          <widget class="QPushButton" name="btn_stopRamping">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Stop ramping</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_8">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QPushButton" name="drawPath">
          <property name="focusPolicy">
           <enum>Qt::ClickFocus</enum>
          </property>
          <property name="text">
           <string>Calculate sequence</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="savePath">
          <property name="text">
           <string>Save sequence</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Matplotlib3dWidget" name="plot" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
        <horstretch>4</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>150</width>
        <height>0</height>
       </size>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PyScientificSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>pyHegel.gui.ScientificSpinBox</header>
  </customwidget>
  <customwidget>
   <class>Matplotlib3dWidget</class>
   <extends>QWidget</extends>
   <header>widgets.Matplotlib3dWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
